### PR TITLE
Merges pilgrims

### PR DESCRIPTION
**Pull request by wow50000**
## About The Pull Request
Deletes some bloat towner roles, allowing towners to be proficient in other fields.
Removed alchemist, alchemist tools and skills have been transferred to barber surgeon
Removed woodcutter, woodcutter tools and skills have been moved to carpenter
Removed fisher, gave hunter some fishing skills and the tools to fish
Nerfed smuggler
Peasant, Seamstress, Drunkard, Minstrel and Smith are untouched
Buffs miner with masonry skill
Buffs noble with stats to help escape the bog.

## Why It's Good For The Game
Due to a declining towner pop and some roles being useless in their specialisations, I have decided to make a PR to merge underperforming roles and buff some of the roles that were very weak relatively. This PR does not touch roles RP roles or roles that are good enough as they are. 

As for the removed roles:
Alchemist was removed due to alchemy alone not being a good enough niche and barber surgeons already being underutilized due to miracles.
Woodcutter was removed due to both woodcutter being inferior to miner and carpenter being useless, thus merging the two.
Fisher was removed and fishing given to hunter so that hunter is now a specialist at gathering food, whereas peasant is the specialist at farming food.
Smuggler was nerfed due to being the only peasant class with combat traits as well as high tier peasant stats and good combat skills and spawning with killer ice, which was originally only one in the entire game. Now smuggler can exist as a side grade to doctor which trades surgery for combat stats and good skills. Making smugglers a more combat focused alchemist.
Peasant was untouched due to it filling the role of "I just want to farm."
Seamstress was untouched for similar reasons as peasant
Drunkard and Minstrel are untouched due to both roles being essentially flavour roles
Smith is untouched due to smith being relatively strong in its own field and able to spec into mining if no miners are around
Miner gets masonry skill to keep in line with the other roles.
Noble gets skills to help them evade capture but also goodlover to reward capturing them ;)

### CODE REVIEW

File: code/modules/jobs/job_types/roguetown/adventurer/types/pilgrim/smuggler.dm [hunk 1]
```diff
@@ -27,10 +27,7 @@
 	backr = /obj/item/storage/backpack/rogue/satchel
 	beltr = /obj/item/rogueweapon/huntingknife/idagger/steel
 	beltl = /obj/item/rogueweapon/huntingknife/idagger/steel
-	backpack_contents = list(/obj/item/reagent_containers/powder/spice = 3, /obj/item/reagent_containers/glass/bottle/mercury = 1, /obj/item/reagent_containers/glass/bottle/rogue/poison = 1, /obj/item/reagent_containers/glass/bottle/rogue/healthpot = 1, /obj/item/reagent_containers/glass/bottle/rogue/manapot = 2,)
-	
-	ADD_TRAIT(H, TRAIT_DODGEEXPERT, TRAIT_GENERIC)
-	ADD_TRAIT(H, TRAIT_GOODRUNNER, TRAIT_GENERIC)
+	backpack_contents = list(/obj/item/reagent_containers/powder/spice = 3, /obj/item/reagent_containers/glass/bottle/mercury = 1, /obj/item/reagent_containers/food/snacks/grown/berries/rogue/poison, /obj/item/reagent_containers/glass/bottle/rogue/healthpot = 1, /obj/item/reagent_containers/glass/bottle/rogue/manapot = 2,)
```
Comment (atomicplan666) [inline code/modules/jobs/job_types/roguetown/adventurer/types/pilgrim/smuggler.dm:30]: WHY REMOVE POISON /rogue/poison

Comment (wow50000) [reply code/modules/jobs/job_types/roguetown/adventurer/types/pilgrim/smuggler.dm:30]: because its an op poison

**Comment by katsukai** [comment]
This is really bad, I honestly wonder how you came to this conclusion.

**Comment by wow50000** [comment]
> This is really bad, I honestly wonder how you came to this conclusion.

what do you want to know ?

**Comment by ThatReformedEngineer** [comment]
The changes aren't too bad besides a few minor nitpicks. Like for example removing the -1 strength on noble is good and tossing in +2 speed is nice, but I don't really see how that counts as a buff that'd really warrant trying out noble. Like how beneficial on average is two speed going to be in the bog?

Each class selected to get merged and combined is pretty sensible, though I could guess some people don't like the carpenter not getting axes 3 from woodcutter.

**Comment by wow50000** [comment]
> The changes aren't too bad besides a few minor nitpicks. Like for example removing the -1 strength on noble is good and tossing in +2 speed is nice, but I don't really see how that counts as a buff that'd really warrant trying out noble. Like how beneficial on average is two speed going to be in the bog?
> 
> Each class selected to get merged and combined is pretty sensible, though I could guess some people don't like the carpenter not getting axes 3 from woodcutter.

I changed noble from a net 0 to a net +4 iirc, honestly losing the axes 3 is rough but most carpenters stay in the town anyways

**Comment by ThatReformedEngineer** [comment]
> I changed noble from a net 0 to a net +4 iirc, honestly losing the axes 3 is rough but most carpenters stay in the town anyways

These are all fair but something that should be noted is how all these buffs just add to pilgrims, no incentive was coded to really be a towner here besides some convivence that applies regardless in the case of Hunter or Miner. The population of towner to pilgrim is still going to be out of balance, unless you got some future PR idea to try addressing that?

**Comment by Elli-Skala** [comment]
a human woodsman has a chance to low roll strength and not be ablt to weild a battle axe.

**Comment by JohnnyJohnJohnson** [comment]
Removed Woodsman and Fisherman? You've murdered me, you've fucking murdered me.

**Pull request closed without merging**